src/bastion-executor/src/load_balancer.rs

```diff
@@ -7,49 +7,102 @@
 use crate::load_balancer;
 use crate::placement;
 use arrayvec::ArrayVec;
+use fmt::{Debug, Formatter};
 use lazy_static::*;
+use once_cell::sync::Lazy;
+use placement::CoreId;
 use std::mem::MaybeUninit;
-use std::sync::atomic::{AtomicUsize, Ordering};
-use std::thread;
-use std::time::Duration;
+use std::sync::atomic::{AtomicBool, AtomicUsize, Ordering};
+use std::sync::RwLock;
+use std::time::{Duration, Instant};
 use std::{fmt, usize};
+use tracing::{debug, error};
+
+const MEAN_UPDATE_TRESHOLD: Duration = Duration::from_millis(200);
 
 /// Stats of all the smp queues.
 pub trait SmpStats {
     /// Stores the load of the given queue.
     fn store_load(&self, affinity: usize, load: usize);
-    /// returns tuple of queue id and load in an sorted order.
+    /// returns tuple of queue id and load ordered from highest load to lowest.
     fn get_sorted_load(&self) -> ArrayVec<[(usize, usize); MAX_CORE]>;
     /// mean of the all smp queue load.
     fn mean(&self) -> usize;
     /// update the smp mean.
     fn update_mean(&self);
 }
-///
-/// Load-balancer struct which is just a convenience wrapper over the statistics calculations.
-#[derive(Debug)]
-pub struct LoadBalancer;
+
+static LOAD_BALANCER: Lazy<LoadBalancer> = Lazy::new(|| {
+    let lb = LoadBalancer::new(placement::get_core_ids().unwrap());
+    debug!("Instanciated load_balancer: {:?}", lb);
+    lb
+});
+
+/// Load-balancer struct which allows us to update the mean load
+pub struct LoadBalancer {
+    /// The number of cores
+    /// available for this program
+    pub num_cores: usize,
+    /// The core Ids available for this program
+    /// This doesn't take affinity into account
+    pub cores: Vec<CoreId>,
+    mean_last_updated_at: RwLock<Instant>,
```

Comment:
You can use a transactional variable of lever.
https://docs.rs/lever/0.1.1-alpha.11/lever/txn/vars/struct.TVar.html

Reply:
or simply Atomic Counter?

Reply:
Just tried to use TVar, but I'm struggling because mean_last_updated_at isn't mut :/, and there doesn't seem to be any atomic instant structure in the stdlib (I can't turn an instant into an int)

Reply:
Ok, I have thread-local idea for this one. I will take a look later.